cmd/config/internal/commands/cmdset.go

```diff
@@ -74,17 +76,24 @@ func initSetterVersion(c *cobra.Command, args []string) error {
 	return nil
 }
 
+
+
 func (r *SetRunner) preRunE(c *cobra.Command, args []string) error {
+	valueSetFromFlag := r.Set.Value != "";
```

Comment:
You should be able to check if a flag has been set, rather than just checking its value.

Reply:
Done, added the helper function isFlagSet, didn't find one in lib